### PR TITLE
Add Proxy to Addon Manager with UI

### PR DESCRIPTION
**Pull request by zackles**
I re-coded the functionnality to use proxy for the Addon Manager.
Proxy is now managable in Addon Manager Options UI with 3 configuration :
- No proxy
- System proxy
- User proxy

This functionnality was develloped beforehand [Proxy settings #63](https://github.com/FreeCAD/FreeCAD-addons/pull/63) and was lost during a merge : 
[Regression : Add Proxy settings to AddonManager in master issue #157](https://github.com/FreeCAD/FreeCAD-addons/issues/157)
[AddonManager does not work behind corporate NTLM-proxy #57](https://github.com/FreeCAD/FreeCAD-addons/issues/57)

Current discussion here : [Proxy internet](https://forum.freecadweb.org/viewtopic.php?f=3&t=38246&start=10#p358970)
My own unresponded question in french : [Proxy internet FR](https://forum.freecadweb.org/viewtopic.php?f=12&t=38255)

---
Thank you for creating a pull request to contribute to FreeCAD! To ease integration, please confirm the following:
- [X] Branch rebased on latest master `git pull --rebase upstream master`
- [ ] Unit tests confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ] Commit message is [well-written](https://chris.beams.io/posts/git-commit/)
- [ ] Commit message includes `issue #<id>` or `fixes #<id>` where `<id>` is the [associated MantisBT](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) issue id if one exists

### CODE REVIEW

File: src/Mod/AddonManager/addonmanager_utilities.py [hunk 1]
```diff
@@ -90,21 +90,38 @@ def urlopen(url):
     timeout = 5
 
     if sys.version_info.major < 3:
-        import urllib2
+        import urllib2
```
Comment (vocx-fc) [inline src/Mod/AddonManager/addonmanager_utilities.py]: Remove trailing whitespace

**Comment by vocx-fc** [comment]
Always provide the forum thread where your changes are discussed.

[Proxy internet](https://forum.freecadweb.org/viewtopic.php?f=3&t=38246&start=10#p358970)

**Comment by zackles** [comment]
> Always provide the forum thread where your changes are discussed.
> 
> [Proxy internet](https://forum.freecadweb.org/viewtopic.php?f=3&t=38246&start=10#p358970)

Sorry :(
And I forgot to mention related issues :
[Regression] Add Proxy settings to AddonManager in master issue #157
AddonManager does not work behind corporate NTLM-proxy #57
Proxy settings #63

**Comment by vocx-fc** [comment]
Your first comment is... weird. I suspect those sentences are the partial commits that you had in your branch before squashing. I suggest writing something more coherent. And yes, adding links where the issues are discussed helps.

**Comment by zackles** [comment]
Yes first time I squash commit so the output is a bit raw and unorganized.

File: src/Mod/AddonManager/addonmanager_utilities.py (hunk 1 above)
Comment (zackles) [reply src/Mod/AddonManager/addonmanager_utilities.py]: made

**Comment by yorikvanhavre** [comment]
Thanks for this contribution! For the next time, there are a few git commands that can be used to change the commit message. But no worries for now, I'll merge.
@zackles please keep an eye on the forum thread in case someone meets problems with your changes (network configurations can always give some headaches on different platforms...)

**Comment by zackles** [comment]
@yorikvanhavre : okay! Like I said in the forum I want to improve to the Addon Manager so I had intention to maintain it.  
I want to thanks @luzpaz and @vocx-fc for their help for my first PR :)

**Pull request merged**